packages/next-dashboard/src/components/layout/Main.js

```diff
@@ -61,75 +62,86 @@ function DashboardLayout({
       window.removeEventListener('resize', noTransition);
     };
   }, []);
+  const context = useContext(DashboardContext);
+  if (!context) {
+    throw new TypeError('Dashboard Layout needs to be in a Dashboard Context');
+  }
+
+  const {
+    getState,
+    theme: { class: themeClass },
+    modalActive,
+  } = context;
+
+  const sidebarActive = getState<boolean>('sidebarActive', false);
+  const sidebarCompact = getState<boolean>('sidebarCompact', false);
+
+  const toggleSidebarActive = () => {
+    context.setState<boolean>('sidebarActive', !sidebarActive);
+  };
+
+  const [scrollOffset, setScrollOffset] = useState<number | null>(null);
+  const [savedScrollOffset, setSavedScrollOffset] = useState<number | null>(
+    null,
+  );
+
+  // When a modal is shown, we want to save the top offset and set position fixed to disable scrolling
+  useEffect(() => {
+    if (scrollOffset === null && !modalActive) return;
```

Comment:
Hm, when is this check necessary/what purpose does it serve?

Reply:
I think it was just to, if scrollOffset and modalActive are "in sync" and disabled, don't touch "setSavedScrollOffset". Just trying to safeguard against edgecases IIRC.

Reply:
Alright, I suppose that's fine :)